prov/util/src/util_attr.c

```diff
@@ -454,8 +454,10 @@ static int fi_progress_level(enum fi_progress progress_model)
 		return 1;
 	case FI_PROGRESS_MANUAL:
 		return 2;
-	case FI_PROGRESS_UNSPEC:
+	case FI_PROGRESS_CONTROL_UNIFIED:
```

Comment:
Did you verify that users of this function still work properly after this change, without needing changes at the calling location?

Reply:
I see the issue. Thanks... pushing fix.

Reply:
All the users of `fi_progress_level()` in main today are inside util_attr.c:
```
(env) ┌─[ec2-user@ip-172-31-22-242]─[1774:23h59]─[~/libfabric]
└─╼ grep -rI fi_progress_level
prov/util/src/util_attr.c:static int fi_progress_level(enum fi_progress progress_model)
prov/util/src/util_attr.c:	if (fi_progress_level(user_attr->control_progress) <
prov/util/src/util_attr.c:	    fi_progress_level(prov_attr->control_progress)) {
prov/util/src/util_attr.c:	if (fi_progress_level(user_attr->data_progress) <
prov/util/src/util_attr.c:	    fi_progress_level(prov_attr->data_progress)) {
```

The four usages are here:
```
	if (fi_progress_level(user_attr->control_progress) <
	    fi_progress_level(prov_attr->control_progress)) {
		FI_INFO(prov, FI_LOG_CORE, "Invalid control progress model\n");
		return -FI_ENODATA;
	}

	if (fi_progress_level(user_attr->data_progress) <
	    fi_progress_level(prov_attr->data_progress)) {
		FI_INFO(prov, FI_LOG_CORE, "Invalid data progress model\n");
		return -FI_ENODATA;
	}
```

We should block FI_PROGRESS_CONTROL_UNIFIED in data progress check here.

It works correctly b/c providers are either specifying FI_PROGRESS_AUTO, (1) or FI_PROGRESS_MANUAL (2) as the level of supported automation.  B/c FI_PROGRESS_CONTROL_UNIFIED returns 3, all providers will accept the threading level without need for modification.